src/Host/Client/Impl/Session/RSessionExtensions.cs

```diff
@@ -23,5 +24,57 @@ public static class RSessionExtensions {
                 await function(evaluation);
             } 
         }
+
+        public static async Task<string> GetRWorkingDirectoryAsync(this IRSession session) {
+            if (session.IsHostRunning) {
+                await TaskUtilities.SwitchToBackgroundThread();
+                try {
+                    using (var evaluation = await session.BeginEvaluationAsync(false)) {
+                        return await evaluation.GetWorkingDirectory();
+                    }
+                } catch (OperationCanceledException) { }
+            }
+            return null;
+        }
+
+        public static async Task<string> GetRUserDirectoryAsync(this IRSession session) {
+            if (session.IsHostRunning) {
+                await TaskUtilities.SwitchToBackgroundThread();
+                try {
+                    using (var evaluation = await session.BeginEvaluationAsync(false)) {
+                        return await evaluation.GetRUserDirectory();
+                    }
+                } catch (OperationCanceledException) { }
+            }
+            return null;
+        }
+
+        public static async Task<string> GetRShortenedPathNameAsync(this IRSession session, string name) {
+            var userDirectory = await session.GetRUserDirectoryAsync();
+            return GetRShortenedPathName(name, userDirectory);
+        }
+
+        public static async Task<IEnumerable<string>> GetRShortenedPathNamesAsync(this IRSession session, IEnumerable<string> names) {
```

Comment:
I don't like the name of the method. Maybe something like `MakeRelativeToUserDirectory`?